src/cell.cpp

```diff
@@ -467,35 +544,40 @@ read_cells(pugi::xml_node *node)
 extern "C" {
   Cell* cell_pointer(int32_t cell_ind) {return global_cells[cell_ind];}
 
-  int32_t cell_id(Cell *c) {return c->id;}
+  int32_t cell_id(Cell* c) {return c->id;}
 
-  void cell_set_id(Cell *c, int32_t id) {c->id = id;}
+  void cell_set_id(Cell* c, int32_t id) {c->id = id;}
 
-  int cell_type(Cell *c) {return c->type;}
+  int cell_type(Cell* c) {return c->type;}
 
-  void cell_set_type(Cell *c, int type) {c->type = type;}
+  int32_t cell_universe(Cell* c) {return c->universe;}
 
-  int32_t cell_universe(Cell *c) {return c->universe;}
+  int32_t cell_fill(Cell* c) {return c->fill;}
 
-  void cell_set_universe(Cell *c, int32_t universe) {c->universe = universe;}
+  int32_t cell_n_instances(Cell* c) {return c->n_instances;}
 
-  int32_t cell_fill(Cell *c) {return c->fill;}
+  int cell_material_size(Cell* c) {return c->material.size();}
 
-  int32_t* cell_fill_ptr(Cell *c) {return &c->fill;}
-
-  int32_t cell_n_instances(Cell *c) {return c->n_instances;}
+  //TODO: off-by-one
+  int32_t cell_material(Cell* c, int i)
+  {
+    int32_t mat = c->material[i-1];
+    if (mat == C_NONE) return F90_NONE;
+    if (mat == MATERIAL_VOID) return MATERIAL_VOID;
```

Comment:
This is starting to look a little messy. Is there really a need to have a separate value indicating "no material was specified" as opposed to "void material". Maybe we should adopt the following behavior:
1. `<cell>` has `material` attribute but no `fill`: single material or distributed materials and `Cell.fill = -1`
2. `<cell>` has `fill` attribute but no `fill`: universe/lattice and `Cell.material = -1`
3. `<cell>` has neither `material` nor `fill`: void material with both `Cell.material = -1` and `Cell.fill = -1`.

If we want to check "Is this material filled with a universe/lattice" (for example, `geometry_aux.cpp:23`), that would just be `Cell.fill >= 0`.

Reply:
Much cleaner, thanks!

Reply:
Note, I went with something slightly different than what you've suggested here.  Case 3 is prevented (at least when reading from XML) by a `fatal_error`.  For case 2, `Cell.material` is just left as an empty vector.